ibis/backends/impala/tests/test_analytic_functions.py

```diff
@@ -12,10 +11,6 @@ def table(mockcon):
 @pytest.mark.parametrize(
     "expr_fn",
     [
-        pytest.param(
-            lambda t: ibis.row_number().over(ibis.window(order_by=t.float_col)),
-            id="row_number",
-        ),
```

Comment:
Is this test bogus in some way? Why did you remove it?

Reply:
`ibis.row_number()` now returns with a deferred since it's not bound to any table yet

calling over returns with a deferred as well since we are trying to bind the window frame if the analytic expression is bound, otherwise we just defer this binding until we pass the expression to a projection